### ISSUE TITLE
Rocketchat Slowness Troubleshooting

### ISSUE THREAD

**Issue by caggles** [BCDevOps/developer-experience#1822]
**Describe the issue**
Rocketchat has been experiencing consistent issues of slowness and problems where users are not always receiving updates on new messages. This seems to coincide with pod restarts or general maintenance, but not always.

**Additional context**
We have tried updating rocketchat versions to alleviate the issue, but it had no effect.

**Definition of done**
- [x] Users no longer experience slowness and notification issues

**Solution**
The property `EXIT_UNHANDLEDPROMISEREJECTION=1` has been added to rocketchat.properties.
A rocketchat-server pod will become partially non-functional after an `UnHandledPromiseRejection` error, which happens when there are certain DB errors, like the primary mongo DB being restarted or a duplicate key error. Clients are still connected to the affected pod, but do not receive updates. Restarting the pod is required. There is not a resource usage problem. Setting `EXIT_UNHANDLEDPROMISEREJECTION` causes the pod to restart when such an error is encountered.

**Comment by caggles** [comment]
I have opened a ticket with RC support. We have been working on troubleshooting it but responsiveness has slowed over the last several days. I have escalated the ticket with our account manager to push for resolution.

**Comment by caggles** [comment]
I'm also working on getting Shelly access to the Rocketchat support site, since apparently she doesn't have the ability to see the tickets I have opened. Once that's completed, I will also work on getting further access set up for the rest of the team.

**Comment by caggles** [comment]
Update: I've had a meeting with RC support on the issue and have come away with some potential action items and some information that they want to help further the investigation on their side. I'm working on getting these action items finished and that information collected and sent away to them.

As for the issue with Shelly being able to access the RC support site, there appears to be some kind of problem in their CRM causing this, but they're working on it.

**Comment by ShellyXueHan** [comment]
https://github.com/RocketChat/Rocket.Chat/issues/21182

**Comment by lukegonis** [comment]
Will revisit this issue after the RC migration

**Comment by ShellyXueHan** [comment]
### Update:

We had another slowness on June 21st on Gold. Currently a restart of RC pods helped to resolve the slowness already. @caggles @IanKWatts 

Here are some action items to plan in soon:
- update HPA to monitor memory usage
- test on doubling DDP rate limit when slowness happens again
- RC upgrade

**Comment by ShellyXueHan** [comment]
@IanKWatts would you update here after trying out the `UnHandledPromiseRejection` configuration change in RC?

**Comment by IanKWatts** [comment]
I found that a rocketchat-server pod will become partially non-functional after an `UnHandledPromiseRejection` error, which happens when there are certain DB errors, like the primary mongo DB being restarted or a duplicate key error.  Clients are still connected to the affected pod, but do not receive updates.  Restarting the pod is required.  There is not a resource usage problem.

The property `EXIT_UNHANDLEDPROMISEREJECTION=1` has been added to rocketchat.properties so that the pod will restart when these types of errors are encountered.

Testing in the Test environment confirmed the problem and verified that the pods will restart when this type of error is encountered.
